File: BackEnd/src/main/java/com/example/backend/data/models/Day.java
package com.example.backend.data.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "days")
public class Day {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @Column(unique = true)
    private Date date;

    private String information;

    @ManyToMany(cascade = CascadeType.DETACH, fetch = FetchType.EAGER)
    @JoinTable(name = "days_lessons")
    private List<LessonAndTime> lessonsAndTimes = new ArrayList<>();

    @Transient
    private String currentUserNote;
}

File: ImpInfApi/ImpInfApi/Data/Models/Day.cs
using System;
using System.Collections.Generic;

namespace ImpInfApi.Data.Entities
{
    public class Day
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Information { get; set; }
        public List<LessonsAndTimes> LessonsAndTimes { get; set; }
        public string CurrentUserNote { get; set; }
    }
}
